<?xml version="1.0"?>

<launch>
  <!-- IP and port on Vicon Windows PC -->
  <arg name="datastream_hostport" default="192.168.1.110:801"/>
  <!-- Fixed vicon world frame -->
  <arg name="vicon_world_frame" default="/vicon/world"/>
  <!-- Vicon target -->
  <arg name="vicon_target" default="/vicon/AZIMUTH/AZIMUTH"/>
  <!-- Serial port -->
  <arg name="serial_port" default="/dev/ttyUSB0"/>
  <!-- baud rate -->
  <arg name="baud" default="115200"/>
      
	<node pkg="vicon_bridge" type="vicon_bridge" name="vicon" output="screen" >
		<param name="stream_mode" value="ServerPush"/>
    <param name="datastream_hostport" value="$(arg datastream_hostport)"/>
		<param name="tf_ref_frame_id" value="$(arg vicon_world_frame)"/>
	</node>
    
    <!-- GPS node --> 
    <node pkg="azimuth" type="simple_listener.py" name="azimuth" output="screen">
        <param name="vicon_target_id" value="$(arg vicon_target)"/>
        <param name="serial_port_handle" value="$(arg serial_port)"/>
        <param name="baud_rate" value="$(arg baud)"/>
    </node>

</launch>
